### PR TITLE
websocket: add fast-path for string input

### PR DESCRIPTION
**Pull request by tsctx**

### CODE REVIEW

**Comment by tsctx** [comment]
Improvement of 10% to 15%

File: lib/web/websocket/frame.js [hunk 1]
```diff
@@ -89,6 +89,49 @@ class WebsocketFrameSend {
...
+    if (bodyLength > maxUnsigned16Bit) {
+      offset += 8 // payload length is next 8 bytes
+      payloadLength = 127
+    } else if (bodyLength > 125) {
+      offset += 2 // payload length is next 2 bytes
+      payloadLength = 126
+    }
+    const head = Buffer.allocUnsafeSlow(offset)
+
+    head[0] = 0x80 /* FIN */ | 0x1 /* opcode TEXT */
```
Comment (KhafraDev) [inline lib/web/websocket/frame.js:119]: please use opcodes.text from the constants

File: lib/web/websocket/frame.js [hunk 2]
```diff
@@ -89,6 +89,49 @@ class WebsocketFrameSend {
...
     return buffer
   }
+
+  /**
+   * @param {string | Uint8Array} body
+   */
+  static createFastTextFrame (body) {
+    const maskKey = generateMask()
+
+    const buffer = typeof body === 'string' ? Buffer.from(body) : body
```
Comment (KhafraDev) [inline lib/web/websocket/frame.js:99]: when can body not be a string?

Comment (tsctx) [reply lib/web/websocket/frame.js:99]: It made this change for #3394, which provides a buffer instead of a string.

Comment (KhafraDev) [reply lib/web/websocket/frame.js:99]: I addressed it in #3394

File: lib/web/websocket/sender.js [hunk 3]
```diff
@@ -31,16 +31,25 @@ class SendQueue {
 
   add (item, cb, hint) {
     if (hint !== sendHints.blob) {
-      const frame = createFrame(item, hint)
       if (!this.#running) {
-        // fast-path
-        this.#socket.write(frame, cb)
+        // TODO(@tsctx): support fast-path for string on running
+        if (hint === sendHints.string) {
```
Comment (KhafraDev) [inline lib/web/websocket/sender.js:36]:
```suggestion
        if (hint === sendHints.text) {
```

**Review by KhafraDev — APPROVED** [review]

**Pull request merged**